compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEOps.td

```diff
@@ -954,7 +959,8 @@ def AMDAIE_NpuDmaWaitOp: AMDAIE_Op<"npu.dma_wait", []> {
   }];
 
   let arguments = (
-    ins Variadic<AMDAIE_AnyAsyncTokenType>:$async_tokens
+    ins Variadic<AMDAIE_AnyAsyncTokenType>:$async_tokens,
+        OptionalAttr<BoolAttr>:$barrier
```

Comment:
Have you though about introducing a separate `barrier` op instead that can be used to break up this and other optimizations? Similar to existing `gpu.barrier` etc.